### PR TITLE
Adds AI Vox Sound Output Circuit

### PR DESCRIPTION
**Pull request by KomradeSpectre**
[Changelogs]:
:cl: TheMythicGhost
:add: New Integrated Circuit Component: AI Vox Sound Circuit
/:cl:

[why]: To give integrated circuitry machines a much better list of words they could provide audible feedback to the user with, or for making suicide drones yell "Ass Blast USA" before they detonate I guess.

### CODE REVIEW

File: code/modules/integrated_electronics/subtypes/output.dm [hunk 1]
```diff
@@ -204,6 +204,20 @@
...
+
+/obj/item/integrated_circuit/output/sound/vox/Initialize()
+	sounds = GLOB.vox_sounds
+	.= ..()
+	extended_desc = list()
+	extended_desc += "The first input pin determines which sound is used. It uses the AI Vox Broadcast word list"
+	extended_desc += ". So either experiment to find words that work, or ask the AI to help in figuring them out"
+	extended_desc += ". The second pin determines the volume of sound that is played"
+	extended_desc += ", and the third determines if the frequency of the sound will vary with each activation."
+	extended_desc = jointext(extended_desc, null)
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/subtypes/output.dm]: Just type the whole thing out in one string, this is just wasted CPU/memory

File: code/modules/integrated_electronics/subtypes/output.dm [hunk 2]
```diff
@@ -204,6 +204,20 @@
...
 	spawn_flags = IC_SPAWN_RESEARCH
 
+/obj/item/integrated_circuit/output/sound/vox
+	name = "ai vox sound circuit"
+	desc = "Takes a sound name as an input, and will play said sound when pulsed. This circuit is often found in AI announcement systems."
+	spawn_flags = IC_SPAWN_RESEARCH
+
+/obj/item/integrated_circuit/output/sound/vox/Initialize()
+	sounds = GLOB.vox_sounds
+	.= ..()
```
Comment (Cyberboss) [inline code/modules/integrated_electronics/subtypes/output.dm]: . = ..() first

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/integrated_electronics/subtypes/output.dm (hunk 1 above)
Comment (KomradeSpectre) [reply code/modules/integrated_electronics/subtypes/output.dm]: I'll do so, but that's how the other sound output circuits are constructed as well so maybe that should be addressed.

**Comment by GunHog** [comment]
So instead of annoying, screaming medi-beepskys, we will have annoying, screaming medi-beepsky AI vox bots? Sign me up! :+1:

File: code/modules/integrated_electronics/subtypes/output.dm [hunk 3]
```diff
@@ -204,6 +204,15 @@
...
 
+/obj/item/integrated_circuit/output/sound/vox
+	name = "ai vox sound circuit"
+	desc = "Takes a sound name as an input, and will play said sound when pulsed. This circuit is often found in AI announcement systems."
+	spawn_flags = IC_SPAWN_RESEARCH
+
+/obj/item/integrated_circuit/output/sound/vox/Initialize()
+	.= ..()
+	sounds = GLOB.vox_sounds
+	extended_desc = "The first input pin determines which sound is used. It uses the AI Vox Broadcast word list. So either experiment to find words that work, or ask the AI to help in figuring them out. The second pin determines the volume of sound that is played, and the third determines if the frequency of the sound will vary with each activation."
```
Comment (JJRcop) [inline code/modules/integrated_electronics/subtypes/output.dm]: Now this could be moved to type definition.

Comment (DaedalusGame) [reply code/modules/integrated_electronics/subtypes/output.dm]: can't, Initialize proc overrides it

Comment (JJRcop) [reply code/modules/integrated_electronics/subtypes/output.dm]: :confused:

**Comment by Spaceinaba** [comment]
now i get even more excuses to delete people for making spam bots

**Comment by TribeOfBeavers** [comment]
Does this work like the actual AI vox and plays the sound for everyone? If so that's going to be heavily abused I'd imagine. You could just dump the bot in maint somewhere and nobody could ever find it to shut it up.

**Comment by KomradeSpectre** [comment]
Reminder, it has a cooldown, and complexity cap. Not to mention it requires metal and power cells galore.

**Comment by Buggy123** [comment]
:+1: but I'd prefer it if we could just take this to its conclusion and get a component that plays arbitrary MIDIs.

**Comment by KomradeSpectre** [comment]
I'd be down for that, but would suggest it have a massive power draw on it.

**Comment by epochayur** [comment]
or have a config-toggle like lobby music or instruments so we don't have to hear MAGIC MISSILE SANIC BEES for 50 minutes straight

**Comment by kevinz000** [comment]
Right I don't think this is a good idea.
If this is done ai vox needs to be logged so we know who is doing what.
Ai vox is unique to the ai for a reason, bots don't need it. Otherwise there needs to be some way of knowing which is which

**Comment by DaedalusGame** [comment]
this isn't real vox, this just adds the vox words to sound chips lol

**Comment by KomradeSpectre** [comment]
It's also locational, which the real AI Vox is not.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**